File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        ts = sum(nums)
        l = 0
        for i in range(n - 1):
            l += nums[i]
            r = ts - l
            if l >= r:
                res += 1
        return res

        

        

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int n = nums.size();
        int c = 0;
        long long ts = 0, ls = 0;
        for (int num : nums){
            ts += num;
        }
        for (int i = 0; i < n -1; i++){
            ls = ls + nums[i];
            long long rs = ts - ls;
            if (ls >= rs){
                c = c + 1;
            }
        }
        return c;
        
    }
};